app/services/invoices/create_invoice_subscription_service.rb

```diff
@@ -133,7 +133,7 @@ def invoicing_reason_for_subscription(subscription)
       # NOTE: upgrading is used as a not persisted reason as it means
       #       one subscription starting and a second one terminating
       return invoicing_reason if invoicing_reason.to_sym != :upgrading
-      return :subscription_terminating if subscription.terminated? && subscription.terminated_at <= timestamp
+      return :subscription_terminating if subscription.terminated? && subscription.terminated_at?(timestamp)
```

Comment:
you can remove the `subscription.terminated?` check if you use this method I believe.

Reply:
:facepalm: thanks!